### PR TITLE
Fix ws close faq

### PR DESCRIPTION
**Pull request by mpaolini**
Update FAQ entry to leverage #754

### CODE REVIEW

File: docs/faq.rst [hunk 1]
```diff
@@ -162,49 +162,43 @@ For example we have an application with two endpoints:
...
-tasks for a user in the :class:`aiohttp.web.Application` instance and
-``cancel()`` them in ``/logout_user`` handler::
+One simple solution is keeping a shared registry of websocket responses
+for a user in the :class:`aiohttp.web.Application` instance and
+call :meth:`aiohttp.web.WebSocketResponse.close` on all of them in ``/logout_user`` handler::
 
     async def echo_handler(request):
 
-        ws = web.WebSocketResponse()
+        ws = aiohttp.web.WebSocketResponse()
```
Comment (asvetlov) [inline docs/faq.rst]: Please keep `web.WebSocketResponse` name.
Canonical way for using aiohttp web is `from aiohttp import web`.

File: docs/faq.rst [hunk 2]
```diff
@@ -162,49 +162,43 @@ For example we have an application with two endpoints:
...
         user_id = authenticate_user(request)
 
-        for task in request.app['websockets'][user_id]:
-            task.cancel()
+        ws_closers = [ws.close() for ws in request.app['websockets'][user_id] if not ws.closed]
 
-        # return response
-        ...
+        # Watch out, this will keep us from returing the response until all are closed
+        ws_closers and await asyncio.wait(ws_closers)
```
Comment (asvetlov) [inline docs/faq.rst]: `aynsio.wait` doesn't raise exceptions, please use `asyncio.gather()`.

File: docs/faq.rst [hunk 3]
```diff
@@ -162,49 +162,43 @@ For example we have an application with two endpoints:
...
+
+        return aiohttp.web.Response(text='OK')
```
Comment (asvetlov) [inline docs/faq.rst]: Name

File: docs/faq.rst [hunk 4]
```diff
@@ -162,49 +162,43 @@ For example we have an application with two endpoints:
...
 
     def main():
         loop = asyncio.get_event_loop()
         app = aiohttp.web.Application(loop=loop)
         app.router.add_route('GET', '/echo', echo_handler)
         app.router.add_route('POST', '/logout', logout_handler)
-        app['websockets'] = defaultdict(set)
+        app['websockets'] = defaultdict(deque)
```
Comment (asvetlov) [inline docs/faq.rst]: Where `deque` comes from and why `set` is not enough?

Comment (mpaolini) [reply docs/faq.rst]: WebSocketResponse is not hashable, that's why I used `collections.deque`

File: docs/faq.rst [hunk 5]
```diff
@@ -162,50 +162,44 @@ For example we have an application with two endpoints:
...
+call :meth:`aiohttp.web.WebSocketResponse.close` on all of them in ``/logout_user`` handler::
 
     async def echo_handler(request):
 
         ws = web.WebSocketResponse()
         user_id = authenticate_user(request)
         await ws.prepare(request)
-        request.app['websockets'][user_id].add(asyncio.Task.current_task())
-
+        request.app['websockets'][user_id][id(ws)] = ws
```
Comment (asvetlov) [inline docs/faq.rst]: `id(ws)` is not reliable: after garbage collection there is a chance to create new `web.WebSocketResponse()` at the same memory address.

**Comment by asvetlov** [comment]
Thanks!

**Pull request merged**